Tests/MCPToolMacrosTests/ToolRegistrationMacroTests.swift

```diff
@@ -0,0 +1,183 @@
+import XCTest
+import SwiftSyntaxMacrosTestSupport
+
+@testable import MCPToolMacros
+
+final class SchemaMacroTests: XCTestCase {
+    
+    func testSchemaMacroWithField() throws {
+        assertMacroExpansion(
+            """
+            @Schema
+            struct UserPreferences {
+                @Field(description: "User's preferred theme", constraint: .options(["light", "dark"]))
+                let theme: String
+                
+                @Field(description: "Enable notifications")
+                let notifications: Bool
+            }
+            """,
+            expandedSource: """
+struct UserPreferences {
+    let theme: String
+    
+    let notifications: Bool
+}
+
+extension UserPreferences: MCP.MCPParameterParsable {
+    public static var inputSchema: MCP.Value {
+        .object([
+            "type": .string("object"),
+            "properties": .object([
+                            "theme": .object(["type": .string("string"), "description": .string("User's preferred theme"), "enum": .array([.string("light"), .string("dark")])]),
+                            "notifications": .object(["type": .string("boolean"), "description": .string("Enable notifications")])
+            ]),
+                    "required": .array([.string("theme"), .string("notifications")])
+        ])
+    }
+
+    public static func parseArguments(_ args: [String: MCP.Value]) -> UserPreferences? {
+        guard
+            let parsedTheme = String(args["theme"] ?? .null),
+                    let parsedNotifications = Bool(args["notifications"] ?? .null)
+        else {
+            return nil
+        }
+
+        return UserPreferences(theme: parsedTheme, notifications: parsedNotifications)
+    }
+}
+""",
+            macros: [
+                "Schema": SchemaMacro.self,
+                "Field": FieldMacro.self
+            ]
+        )
+    }
+    
+    func testSchemaMacroWithOptionalField() throws {
+        assertMacroExpansion(
+            """
+            @Schema
+            struct UserProfile {
+                @Field(description: "User name")
+                let name: String
+                
+                @Field(description: "User email")
+                let email: String?
+            }
+            """,
+            expandedSource: """
+struct UserProfile {
+    let name: String
+    
+    let email: String?
```

Comment:
In `testSchemaMacroWithOptionalField` expanded source, `guard let parsedEmail = String(args["email"] ?? .null)` makes the optional field effectively required at runtime. Suggest changing the macro to generate optional handling: `let parsedEmail = String(args["email"] ?? .null)` (no guard) and update the expected expansion accordingly.